### PR TITLE
Option for URL building (factory class)

### PR DESCRIPTION
**Pull request by goofballLogic**

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -1,18 +1,37 @@
 import { firstOrDefaultValue, firstOrDefaultContent } from "/js/parsing.js";
 
 const projectId = "99b0b3b0-4838-0051-3d57-8af72f55e8a0";
-const apiItems = "https://deliver.kontent.ai/{0}/items{1}";
-const apiItem = "https://deliver.kontent.ai/{0}/items/{1}";
+const apiItems = `https://deliver.kontent.ai/${projectId}/items{1}`;
```
Comment (YorgunKelebek) [inline js/app.js]: Need to remove {1} at the end?

File: js/app.js [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
+        const searchParams = working.searchParams;
+        if (this.includeTotalCount) searchParams.set("includeTotalCount", this.includeTotalCount);
+        if (this.limit) searchParams.set("limit", this.limit);
+        if (this.orderByElementProp) searchParams.set("order", `elements.{${this.orderByElementProp}}`);
+        if (this.blogTagsContains) searchParams.set("elements.blog_tags[contains]", this.blogTagsContains);
+        return working;
+    }
+}
+
+const itemsURLBuilder = new URLBuilder(apiItems);
```
Comment (YorgunKelebek) [inline js/app.js]: Consider renaming - it can also be used for other API urls, so maybe something like kontentApiBuilder

File: js/app.js (hunk 1 above)
Comment (goofballLogic) [reply js/app.js]: true!

File: js/app.js (hunk 2 above)
Comment (goofballLogic) [reply js/app.js]: yes you are right.

**Comment by goofballLogic** [comment]
Closing in favour of the other one

**Comment by YorgunKelebek** [comment]
I think this is still a better option...

**Pull request closed without merging**